src/ActivityHeartbeatingCancellation/MyActivities.cs

```diff
@@ -0,0 +1,37 @@
+using Microsoft.Extensions.Logging;
+using Temporalio.Activities;
+using Temporalio.Exceptions;
+
+namespace TemporalioSamples.ActivityHeartbeatingCancellation;
+
+public static class MyActivities
+{
+    [Activity]
+    public static async Task FakeProgressAsync(int sleepIntervalMs = 1000)
+    {
+        try
+        {
+            // Allow for resuming from heartbeat
+            var startingPoint = ActivityExecutionContext.Current.Info.HeartbeatDetails.Any()
+                ? await ActivityExecutionContext.Current.Info.HeartbeatDetailAtAsync<int>(0)
+                : 1;
+
+            ActivityExecutionContext.Current.Logger.LogInformation("Starting activity at progress: {StartingPoint}", startingPoint);
+
+            for (var progress = startingPoint; progress <= 100; ++progress)
+            {
+                ActivityExecutionContext.Current.CancellationToken.ThrowIfCancellationRequested();
+                ActivityExecutionContext.Current.Logger.LogInformation("Progress: {Progress}", progress);
+                await Task.Delay(sleepIntervalMs);
+                ActivityExecutionContext.Current.Heartbeat(progress);
+            }
+
+            ActivityExecutionContext.Current.Logger.LogInformation("Fake progress activity completed");
+        }
+        catch (OperationCanceledException e) when (e.InnerException is CancelledFailureException)
```

Comment:
```suggestion
        catch (OperationCanceledException)
```

I think demonstrating just catching the cancel is good enough and probably better practice than relying on the cause